domains/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -0,0 +1,1093 @@
+        BridgeBlocked,
+        /// Slot ordering of light client update is invalid
+        InvalidSlotsOrdering,
+        /// Sync committee signed the attested header is not recognized
+        UnrecognizedSyncCommittee,
+        /// Light client update does not contain any new information
+        NonRelevantUpdate,
+        /// Light client update does not contain enough information to be considered
+        NotApplicableUpdate,
+        /// Light client update should not contain finalized header
+        NonEmptyFinalizedHeader,
+        /// Light client update should not contain sync committee
+        NonEmptySyncCommittee,
+        /// Next sync committee in light client does not match stored committee for that period
+        NextSyncCommitteeMismatch,
+        /// Finalized header's sync committee period does not match store's current period
+        FinalizedPeriodMismatch,
+        /// Unable to store non sequential sync committee update
+        NonSequentialSyncCommitteeUpdate,
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+    /// Historical execution headers
+    #[pallet::storage]
+    pub(super) type ExecutionHeaders<T: Config> =
+        CountedStorageMap<_, Identity, H256, ExecutionHeaderOf, OptionQuery>;
+
+    /// Mapping of count -> Execution header hash. Used to prune older headers
+    #[pallet::storage]
+    pub(super) type ExecutionHeadersMapping<T: Config> =
+        StorageMap<_, Identity, u64, H256, ValueQuery>;
+
+    /// Historical sync committees till the sync committee of latest finalized header.
+    #[pallet::storage]
+    pub(super) type SyncCommittees<T: Config> =
+        CountedStorageMap<_, Identity, u64, SyncCommitteeOf, ValueQuery>;
+
+    /// Genesis validators root
+    #[pallet::storage]
+    pub(super) type GenesisValidatorsRoot<T: Config> = StorageValue<_, H256, ValueQuery>;
+
+    /// Latest finalized header observed
+    #[pallet::storage]
+    pub(super) type LatestFinalizedHeaderState<T: Config> =
+        StorageValue<_, FinalizedHeaderState, ValueQuery>;
+
+    /// Latest execution header observed
+    #[pallet::storage]
+    pub(super) type LatestExecutionHeaderState<T: Config> =
+        StorageValue<_, ExecutionHeaderState, ValueQuery>;
+
+    /// Latest sync committee period observed
+    #[pallet::storage]
+    pub(super) type LatestSyncCommitteePeriod<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+    /// Boolean signifying whether the bridge is blocked or not
+    #[pallet::storage]
+    pub(super) type Blocked<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+    #[pallet::genesis_config]
+    #[derive(Default)]
+    pub struct GenesisConfig {
+        pub initial_sync: Option<InitialSyncOf>,
+    }
+
+    #[cfg(feature = "std")]
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig {
+        fn build(&self) {
+            log::info!(
+                target: "ethereum-beacon-client",
```

Comment:
We typically add a `runtime` prefix for the target in pallets `runtime::ethereum-beacon-client`.